### ISSUE TITLE
Training on a small sample on GPU

### ISSUE THREAD

**Issue by adtyashgl** [yhenon/keras-frcnn#54]
Hi,
I am a complete newbie to machine learning and would appreciate some input/advice.
I am trying to train keras-frcnn to detect parking lots from satellite images. 
I have collected around 500 sample images (all of size 1920 * 720 pixels) and have labelled the parking lots as dictated by simple_parser.py format.

In order to test whether my labelling is correct or indeed if the detection would work, I have started training on a small set of 28 images of size 1920 * 720 pixels.

I have setup an amazon AWS GPU machine and have installed CUDA , cuDNN and tensorflow GPU , keras and python 2.7

I started my training on the batch of 28 and it has currently completed 48 epochs (24 hours+) and is still continuing. I was able to capture the end of the 48th epoch 
Loss Detector classifier: 0.00371048863394
Loss Detector regression: 0.00142121276243
Elapsed time: 2149.8460629
Epoch 48/2000

I am not sure if the training should take this long on a batch of 28 images? I was wondering if someone can advise as to whether this is expected or if I am doing something wrong (perhaps the image size is too large)

Would appreciate any input. 

Thanks

**Comment by yhenon** [comment]
Hey,


> all of size 1920 * 720 pixels
did you change `im_size` in `config.py`? if not your images will be resized to have their smallest size = 600 pixels

> Elapsed time: 2149.8460629

Epochs are hard coded to have length 1000 (see line 137 of train_frcnn.py). If you didnt change that, then your epochs are 50 images, not 1000. If you did change this - that seems very slow. Make sure tensorflow is indeed finding your GPU.

> Loss Detector classifier: 0.00371048863394
> Loss Detector regression: 0.00142121276243


These are very low losses - your network has converged (and likely has extreme overfitting on these 50 images).

**Comment by adtyashgl** [comment]
Hi @yhenon ,
Thanks for your response. No  I didnt change im_size in config.py. I was under the impression that the training re-sizes the images to be of 600 pixels and therefore do not touch that.

> Epochs are hard coded to have length 1000. 

Again I didn't change that. 

> These are very low losses - your network has converged (and likely has extreme overfitting on these 50 images).

Understood, I will try the test now. If it doesnt work, I will re-train with changing the image size and the number of epochs to 28.
Thanks your time.

**Comment by jyotikr** [comment]
Hi @yhenon 

I am doing very similar thing as @aditashgl and I came across this post. I have something to clarify.

When you said "did you change im_size in config.py" and said if not done then training re-sizes the image to smallest size of 600 pixels. What do you mean by this? 

does im_size is total pixels in an image? i.e if I have all image of size 100*100 pixels then im_size should be 10000 if I dont want training to re-size it?

Thanks in advance for your help on this. 

Regards

**Comment by HuangBo-Terraloupe** [comment]
Hi @yhenon 
Do you have some experience what the size of image patches should be?
When we train on other dataset, is it better to crop the image as the same size of VOC dataset ?
Because image size, region proposals and anchor size, everything are related.
